src/client/Microsoft.Identity.Client/AuthenticationResult.cs

```diff
@@ -263,5 +268,46 @@ public string CreateAuthorizationHeader()
                 _authenticationScheme?.AuthorizationHeaderPrefix ?? TokenType,
                 AccessToken);
         }
+
+#if !NETSTANDARD1_3
+        /// <summary>
+        /// Returns a <See cref= "ClaimsPrincipal"/> object containing the claims extracted from the ID Token.
+        /// </summary>
+        public ClaimsPrincipal GetIdTokenClaims()
+        {
+            if (string.IsNullOrEmpty(IdToken))
+                return null;
+
+            string[] idTokenSegments = IdToken.Split(new[] { '.' });
+
+            if (idTokenSegments.Length < 2)
```

Comment:
>< 2 [](http://example.com/codeflow?start=39&length=3)

I think it should be 3, and the message should be that it is specifically a JWT format, which has three base64url-encoded strings, separated by dots.

Reply:
This splitting logic can be consolidated. The code is there from day 1 of ADAL probably :). Possibly to take into account v1 IDTokens, which are not signed.

Reply:
This is the same logic we use to parse the idtoken to create idtoken object